nms/UpdatedMetaProvider/src/net/ess3/nms/updatedmeta/BukkitSpawnerBlockProvider.java

```diff
@@ -0,0 +1,31 @@
+package net.ess3.nms.updatedmeta;
+
+import net.ess3.nms.SpawnerBlockProvider;
+import org.bukkit.block.CreatureSpawner;
+
+public class BukkitSpawnerBlockProvider extends SpawnerBlockProvider {
+    @Override
+    public void setMaxSpawnDelay(CreatureSpawner spawner, int delay) {
+        spawner.setMaxSpawnDelay(delay);
+    }
+
+    @Override
+    public void setMinSpawnDelay(CreatureSpawner spawner, int delay) {
+        spawner.setMinSpawnDelay(delay);
+    }
+
+    @Override
+    public boolean tryProvider() {
```

Comment:
`tryProvider` should be implemented by the abstract provider class and should attempt to call one of the provider's methods. Otherwise we don't know for certain that this provider actually does what it's supposed to in a consistent fashion.

Reply:
Difficult to do in this circumstance, since we do not have a `CreatureSpawner` instance to call the provider's methods on. For the other providers, it's easier to do this since it only uses `ItemStack`, `EntityType` and `Material` classes, which are enums or easily instantiated. That's why I opted to implement `tryProvider` in the concrete class rather than in the abstract one. It's definitely different, but I don't think it's bad. The way it's set up, if the methods required from the API do not exist, we safely assume that this is MC 1.8-1.11, where we need NMS (old versions are not going to change). Otherwise, we can rely on the API (the implementation may change, but it should not concern us).